internal/util/dataplane/mode.go

```diff
@@ -1,12 +1,32 @@
 package dataplane
 
+type DBMode string
```

Comment:
I'd suggest adding the following constructor that would be used in places where we accept an arbitrary string and want to convert it to a strongly typed enum to ensure data integrity:
```go
func NewDBMode(mode string) (DBMode, error) { 
  switch mode {
    case DBModeOff, "": 
      return DBModeOff, nil
    case DBModePostgres:
      return DBModeOn, nil
    default:
      return "", fmt.Errorf("unknown DBMode passed: %s", mode)
  }
}
```

Reply:
What would we gain with this? In Go there's no way to enforce the enum to not use the underlying type so anyone could still use a string literal where `DBMode` would be required. If there's not validation logic here or any other mechanism that would prevent this and I don't see the added benefit here.

Reply:
We'd gain a common way to instantiate a valid instance of this enum. With the current approach, I agree it's still possible to abuse it because you may create an instance with just `DBMode("arbitrary-string")`. 

There's a way to make it impossible though: we could disallow defining invalid values outside of the origin package by using a [struct-based enum](https://threedots.tech/post/safer-enums-in-go/):

```go
type DBMode struct {
	mode string
}

var (
	DBModeOff      = DBMode{"off"}
	DBModePostgres = DBMode{"postgres"}
)

func NewDBMode(mode string) (DBMode, error) {
	switch mode {
	// Both explicit "off" and an empty string are treated as "off".
	case DBModeOff.mode, "":
		return DBModeOff, nil
	case DBModePostgres.mode:
		return DBModePostgres, nil
	}
	return DBMode{}, fmt.Errorf("unknown db mode %q", mode)
}
```
This way, you can only create a zero value of the struct, which (if needed) can be verified by packages depending on it and rejecting such values.

Reply:
> This way, you can only create a zero value of the struct

I initially wanted to write that this empty default being implicitly `off` feels like a trap for user of such API but I believe that since we're going to use this in the context of reading the value from Kong Gateway's config this perhaps would be fine.

One thing we'll need to add to this API is the possibility to compare 2 values, since those being `struct`s withh an unexported field, make it rather non trivial if impossible.

Reply:
> I initially wanted to write that this empty default being implicitly off feels like a trap for user of such API but I believe that since we're going to use this in the context of reading the value from Kong Gateway's config this perhaps would be fine.

Yeah, this is just a mapping between the Kong and KIC domain. As unfortunate as it is, an empty string is a valid value to tell it's DB-less mode in Kong.

> One thing we'll need to add to this API is the possibility to compare 2 values, since those being structs withh an unexported field, make it rather non trivial if impossible.

Not really. As long as the type is composed of comparable types it can be safely compared with `==` operator - which is the case here. 

> Struct types are comparable if all their field types are comparable. Two struct values are equal if their corresponding non-[blank](https://go.dev/ref/spec#Blank_identifier) field values are equal. The fields are compared in source order, and comparison stops as soon as two field values differ (or all fields have been compared).
>
>Source: https://go.dev/ref/spec#Comparison_operators

Reply:
Added `NewDBMode()` with validation.